Source/Core/Core/PatchEngine.cpp

```diff
@@ -223,7 +223,8 @@ static bool IsStackSane()
     return false;
 
   // Check the link register makes sense (that it points to a valid IBAT address)
-  return PowerPC::HostIsInstructionRAMAddress(PowerPC::HostRead_U32(next_SP + 4));
+  u32 address = PowerPC::HostRead_U32(next_SP + 4);
+  return PowerPC::HostIsInstructionRAMAddress(address) && PowerPC::HostRead_U32(address);
```

Comment:
Please use explicit boolean operators with arithmetic values (in this case `!= 0`).